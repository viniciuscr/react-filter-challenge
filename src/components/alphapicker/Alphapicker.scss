@import "../../colors.scss";
.page-link {
  color: $gray;
  text-transform: uppercase;
  font-weight: 500;
  border-left: 0;

}
.page-item:first-of-type .page-link{
    border-left: 1px solid $light-gray;
    border-right: 2px solid $light-gray !important;
}
.page-link{
  cursor: pointer;
}

.picked{
  background-color: $light-gray;
}

.pagination {
  /* we want the alphapicker to break to a new line when there is no space */
  flex-wrap: wrap;
}