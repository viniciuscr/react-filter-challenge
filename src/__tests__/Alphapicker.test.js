import React from "react";
import { mount } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Alphapicker from "../components/alphapicker";

configure({ adapter: new Adapter() });

describe("Alphapicker tests", () => {});

it("should render as expected", () => {
  const wrapper = mount(
    <Alphapicker pickFunc={() => {}} picks={["b", "3", "bar"]} />
  );
  expect(wrapper.html()).toMatchSnapshot();
  wrapper.unmount();
});

it("should have the pick 'All' by default ", () => {
  const pickFunc = jest.fn();
  const wrapper = mount(
    <Alphapicker pickFunc={pickFunc} picks={["b", "3", "bar"]} />
  );

  expect(wrapper.findWhere(node => node.key() === "all").exists()).toBeTruthy();

  wrapper.unmount();
});

it("should call pickFunc when a pick is clicked", () => {
  const pickFunc = jest.fn();
  const wrapper = mount(
    <Alphapicker pickFunc={pickFunc} picks={["b", "3", "bar"]} />
  );
  wrapper
    .findWhere(node => node.key() === "bar")
    .find("button")
    .simulate("click");

  expect(pickFunc).toHaveBeenCalledWith({ name: "bar" });

  wrapper.unmount();
});

it("should call pickFunc with empty string and unselect any picked when clicked on All", () => {
  const pickFunc = jest.fn();
  const wrapper = mount(
    <Alphapicker pickFunc={pickFunc} picks={["b", "3", "bar"]} />
  );
  wrapper
    .findWhere(node => node.key() === "all")
    .find("button")
    .simulate("click");

  expect(pickFunc).toHaveBeenCalledWith({ name: null });
  expect(wrapper.html()).toMatchSnapshot();

  wrapper.unmount();
});
