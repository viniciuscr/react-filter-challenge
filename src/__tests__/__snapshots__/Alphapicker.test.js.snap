// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should call pickFunc with empty string and unselect any picked when clicked on All 1`] = `"<nav aria-label=\\"Alphapicker\\"><ul class=\\"pagination\\"><li class=\\"page-item\\"><button class=\\"page-link\\">All</button></li><li class=\\"page-item\\"><button class=\\"page-link \\">b</button></li><li class=\\"page-item\\"><button class=\\"page-link \\">3</button></li><li class=\\"page-item\\"><button class=\\"page-link \\">bar</button></li></ul></nav>"`;

exports[`should render as expected 1`] = `"<nav aria-label=\\"Alphapicker\\"><ul class=\\"pagination\\"><li class=\\"page-item\\"><button class=\\"page-link\\">All</button></li><li class=\\"page-item\\"><button class=\\"page-link \\">b</button></li><li class=\\"page-item\\"><button class=\\"page-link \\">3</button></li><li class=\\"page-item\\"><button class=\\"page-link \\">bar</button></li></ul></nav>"`;
